cluster-autoscaler/core/equivalence_groups.go

```diff
@@ -67,7 +67,7 @@ func groupPodsBySchedulingProperties(pods []*apiv1.Pod) map[equivalenceGroupId][
 
 		matchingFound := false
 		for _, g := range equivalenceGroupsByController[controllerRef.UID] {
-			if reflect.DeepEqual(pod.Labels, g.representant.Labels) && apiequality.Semantic.DeepEqual(pod.Spec, g.representant.Spec) {
```

Comment:
Is this change backwards-compatible with k8s versions < 1.21?

Reply:
Usually CA versions will map 1:1 with k8s versions. But you're gonna hit the same issue if you're using say CA 1.19 on a K8s 1.21 cluster. The core of the issue is the SA TokenVolume behaviour.

Reply:
so yes backwards compatible but ppl don't mix and match k8s and CA versions

Reply:
That makes sense (although I bet there *are* some folks out there who keep their k8s clusters running the same version while eagerly getting new cluster-autoscaler versions — this mostly works IMO so we wanna be clear if this change literally breaks when running on k8s 1.19, for example)